Source/WTF/wtf/Ref.h

```diff
@@ -139,6 +139,15 @@ class Ref {
     friend Ref adoptRef<T>(T&);
     template<typename X, typename Y> friend class Ref;
 
+    template<typename X, typename Y, typename W, typename Z>
```

Comment:
nit: The pattern is T, U for lhs and X, Y for rhs elsewhere, would be nice to keep the consistency.

Reply:
T conflicted when I write this for Kiara. See `typename X, typename Y` above.

Reply:
Right, so you can just do `template <typename X, typename Y> friend bool operator==(const Ref&, const Ref<X, Y>&);`

Reply:
to be clear the X Y above is so that every Ref instantiation is friended with every other Ref instantiation.

Reply:
Was getting build errors for the first Ref not being templated. Left this as is, but open to suggestions on how to resolve.